### PR TITLE
feat(web): [R5] add career evidence map UI

### PR DESCRIPTION
**Pull request by ebarti**
## What

Delivers Phase E2 from PR #263: the Career Evidence Map UI.

- Adds `/evidence-map` with Operations-owned query keys/hooks, a typed API client port, and TanStack Router search state for entry/job filters.
- Renders proof-point stories, freshness, resume usage links, requirement-fit links, gaps, and reusable-story summaries from the E1 `GET /v1/evidence-map` read model.
- Adds entry points from the app nav, Profile editor, and Jobs detail drawer.
- Extends MSW/QA fixtures with projection-backed evidence usage data, plus Storybook populated/loading/empty/error states.
- Documents the user flow and adds an Evidence Map Smoke QA checklist entry.

## Why

The plan requires the evidence map to be inspectable before interview preparation starts. This phase makes the E1 read model user-visible without adding new facts or write paths: usages and gaps are read-only links back to existing artifact and job surfaces.

## Validation

- `corepack pnpm --filter @jobhunter/web build`
- `corepack pnpm web:check`
- `corepack pnpm --filter @jobhunter/web test src/contexts/operations/queryKeys.test.ts src/contexts/operations/hooks/useEvidenceMapQuery.test.ts src/views/evidence-map/EvidenceMapView.test.tsx src/views/evidence-map/EvidenceMapView.a11y.test.tsx`
- `corepack pnpm --filter @jobhunter/web test-d`
- `corepack pnpm --filter @jobhunter/api-client check`
- `corepack pnpm --filter @jobhunter/web test`
- `corepack pnpm --filter @jobhunter/web e2e -- tests/evidence-map.spec.ts`
- `corepack pnpm check`
- `corepack pnpm docs:build`
- `corepack pnpm web:storybook:build`
- `corepack pnpm web:storybook:test`
- `git diff --check`

### CODE REVIEW

File: apps/web/src/contexts/operations/evidenceMapKeys.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type { TenantId } from "@jobhunter/domain-types";
+
+export const evidenceMapKeys = {
```
Comment (ebarti) [inline apps/web/src/contexts/operations/evidenceMapKeys.ts:3]: PR Reviewer - Claude

Severity: Medium
Gate: Pass
Evidence: `git grep evidenceMapKeys` returns only evidenceMapKeys.ts, hooks/useEvidenceMapQuery.ts, queryKeys.ts, queryKeys.test.ts, and routes/evidence-map.tsx — never any `contexts/*/handlers.ts`. Meanwhile the events that rebuild E1's `evidence_usage_projections` all have handlers that invalidate other contexts only: `bulletProvenanceRecordedHandler` (contexts/materials/handlers.ts:72), `resumeApprovedHandler` (materials/handlers.ts:25), `jobScoredHandler` (contexts/scoring/handlers.ts:7), `profileUpdatedHandler`/`profileImportedHandler` (contexts/profile/handlers.ts:13,17). The full web suite passing (915 tests) confirms no test covers map SSE refresh. The header meta also never renders the response `generatedAt`, so staleness is invisible.
Reproduction: Reasoned from code + git grep. Not run: opening the live map and tailoring in another tab needs a server (out of review scope).

The Career Evidence Map query is never invalidated by the SSE invalidation router. Phase E2 scope in docs/plans/2026-07-05-evidence-map-interview-prep-plan.md explicitly requires registering invalidation handlers so the map refreshes on `BulletProvenanceRecorded` / `JobScored` / `ProfileUpdated`, and Owner Decision #1 chose the projection precisely so "the map is SSE-fresh." As written, an open map shows stale usages/gaps after a resume is tailored, a job is scored, or the profile is edited/imported until the route remounts or the window refocuses. This defeats the stated SSE-freshness design goal for an audit surface.

Required fix: Add `invalidate(evidenceMapKeys.all(event.tenantId))` to the `BulletProvenanceRecorded`, `ResumeApproved`, `JobScored`, `ProfileUpdated`, and `ProfileImported` handlers (they already import Operations key factories, so this is in-pattern), and add a parity/regression test asserting those events invalidate the evidence-map keys. Consider also surfacing `generatedAt` in the header so freshness is visible.

File: apps/web/src/views/evidence-map/EvidenceMapView.tsx [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+}
+
+function UsageGroup({
+  title,
+  usages,
+}: {
+  readonly title: string;
+  readonly usages: readonly EvidenceUsageRef[];
+}) {
+  if (!usages.length) {
```
Comment (ebarti) [inline apps/web/src/views/evidence-map/EvidenceMapView.tsx:126]: PR Reviewer - Claude

Severity: Medium
Gate: Pass
Evidence: `UsageGroup` returns `null` when `usages` is empty (lines 126-128), and `EvidenceDetail` renders only the groups that are non-empty (lines 236-238). An `EvidenceMapEntry` whose `resumeUsages`, `requirementUsages`, and `coverageUsages` are all empty therefore shows story + skills and no usage sections at all — with no "never used" statement. Contrast the story block, which honestly renders "Not recorded" per field (lines 204/208/212). The only zero-usage signal is the "0 uses" badge in the list (line 170), not in the detail where the user inspects.
Reproduction: Reasoned from code; the `Empty` Storybook story exercises the all-empty map, but no fixture covers a single entry with zero usages in a populated map.

This is an audit-surface honesty gap. A newly added proof point, or evidence for skills no target job has demanded yet, is exactly the "you have this but never use it" insight the evidence map exists to surface (plan goal: "inspect where used"). Omitting the sections silently hides that state instead of stating it.

Required fix: Render an explicit empty state per usage group (e.g. "Not used in any resume yet" / "No requirement-fit history recorded" / "No coverage recorded"), matching the story section's "Not recorded" treatment, and add a component test asserting an entry with zero usages shows the explicit not-used copy.

File: apps/web/src/views/evidence-map/EvidenceMapView.tsx [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+        <p className="meta">{entryKindLabel(entry)}</p>
+        <h2 id="evidence-detail-title">{entry.title}</h2>
+        <div className="evidence-tags">
+          <span className={`tag ${tagTone(freshness.evidenceStrength)}`}>
+            {freshness.evidenceStrength ?? "unrated"}
+          </span>
+          <span className={freshness.userConfirmed ? "tag ok" : "tag warn"}>
+            {freshness.userConfirmed ? "confirmed" : "unconfirmed"}
+          </span>
+          <span className="tag muted">{compactDate(freshness.evidenceDateRange)}</span>
```
Comment (ebarti) [inline apps/web/src/views/evidence-map/EvidenceMapView.tsx:194]: PR Reviewer - Claude

Severity: Medium
Gate: Pass
Evidence: The E1 read model records recency/version provenance on every usage — `EvidenceUsageRef.generation`, `scoreVersion`, and `occurredAt` — and on the entry — `EvidenceFreshness.lastUsedAt` and `claimConfidence` (packages/contracts/src/schemas.ts:1946-1971). None are rendered. The freshness header here (lines 187-195) shows only strength, confirmed, and date range; `resumeLinkLabel`/`requirementLinkLabel` (lines 79-93) emit label text plus a status badge with no timestamp or generation.
Reproduction: Reasoned from code; confirmed by reading the full EvidenceMapView.tsx — no `occurredAt`/`generation`/`scoreVersion`/`lastUsedAt` reference and no `title=` tooltip anywhere.

For an audit surface the "when" must be inspectable, not just a bare status badge. The plan's freshness invariant is explicitly "evidence date + confirmation + last-used," and a usage that links to an artifact/job without showing which generation or when it was recorded makes it hard to tell a current usage from a stale one.

Required fix: Surface `freshness.lastUsedAt` in the detail header and render each usage's `occurredAt` and `generation` (or `scoreVersion` for requirement usages) on the usage row, and extend the component test to assert the recorded generation/timestamp is shown.

File: apps/web/src/views/evidence-map/EvidenceMapView.tsx [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+        params={{ artifactId: usage.artifactId }}
+        to="/artifacts/$artifactId"
+      >
+        <span>{resumeLinkLabel(usage)}</span>
+        <span className="tag muted">artifact</span>
+      </Link>
+    );
+  }
+  return (
+    <Link className="evidence-usage-link" params={{ jobId: usage.jobKey }} to="/jobs/$jobId">
```
Comment (ebarti) [inline apps/web/src/views/evidence-map/EvidenceMapView.tsx:110]: PR Reviewer - Claude

Severity: Low
Gate: Pass
Evidence: `UsageLink` deep-links only `resume_bullet` usages to `/artifacts/$artifactId` (lines 95-108); every other usage — including `skill_coverage` — falls through to the `/jobs/$jobId` branch here (line 110). But coverage usages carry a recorded `artifactId` (the QA fixture's Python skill coverage sets `artifactId: "artifact-resume-1"`, `generation: 3`, `keyword: "Python"` in apps/web/src/test/fixtures/projections.ts), so the resume where the skill was actually covered is discarded and the link points at the job instead.
Reproduction: Reasoned from code + the shipped fixture (skill entry `skill:backend:0` coverageUsage has an artifactId that the UI ignores).

The read model records artifact-level coverage provenance; routing it to the job loses precision the map is meant to expose ("Coverage history" should point at the resume that covered the keyword).

Required fix: When a coverage (or requirement) usage has a non-null `artifactId`, link to `/artifacts/$artifactId` (keep the job link as fallback when it is null).

File: apps/web/src/views/evidence-map/EvidenceMapView.tsx [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+                <EvidenceEntryButton
+                  entry={entry}
+                  key={entry.entryId}
+                  search={search}
+                  selected={entry.entryId === selectedEntryId}
+                />
+              ))}
+            </ul>
+          ) : (
+            <Empty title="No evidence entries match." />
```
Comment (ebarti) [inline apps/web/src/views/evidence-map/EvidenceMapView.tsx:352]: PR Reviewer - Claude

Severity: Low
Gate: Pass
Evidence: This branch renders "No evidence entries match." whenever `filteredEntries` is empty (line 352), which includes a profile that has no evidence at all with no active search/job filter — "match" implies a filter is in effect. The `Empty` Storybook story exercises exactly this zero-data case. Relatedly, the header meta shows `"loading"` whenever `evidenceMap.data` is undefined (the CardHeader `meta` prop), so an errored load also reads as "loading" while the error banner is shown.
Reproduction: Reasoned from code; the `Empty` and `Error` stories reproduce both states.

Minor honesty/clarity: a genuinely empty evidence map should say so plainly rather than implying a filter hid results, and an error should not read as "loading."

Required fix: Distinguish "No evidence recorded yet." (no data, no filters) from "No evidence entries match your search/filter." (filters active), and set the header meta to an error/empty state rather than "loading" when the query has settled.

**Review by ebarti — COMMENTED** [review]
## PR Reviewer - Claude — R5 E2 Career Evidence Map UI

**Gate: PASS** (no Blocker/High). Reviewed head `5d8be06f` against base `feat/r5-e1-evidence-usage-index` (#279, `3a8d72a5`). Base and stack (#263←#276←#279←**#283**←#285) verified correct — reviewed only this PR's diff.

The core audit-honesty invariant holds: every rendered linkage reads recorded fields straight off E1's `EvidenceMapResponse` (artifactId, jobKey, requirementId, generatedTextPreview, coverageState). There is **no** client-side synthesis, keyword-matching, or heuristic join presented as recorded evidence — the search/job filters only narrow which recorded entries display. Deep links resolve to real routes (`/artifacts/$artifactId`, `/jobs/$jobId`), all three entry points (NavBar, ProfileEditor, JobDetailDrawer) use `<Link>` (no `<a href>` full-reload regression from #291), reads go through Operations hooks only, query keys follow `["tenant", tenantId, "evidence-map", …]` and are re-exported through `operations/queryKeys.ts`, and no view-level `useQuery`/`apiClient`/`localStorage`/`EventSource` usage. Docs, stories (5 states), a11y test, e2e spec, and MSW fixtures are all present.

### Findings

| Severity | Thread | Finding |
| --- | --- | --- |
| Medium | [r3525823785](https://github.com/ebarti/JobHunter/pull/283#discussion_r3525823785) | Map is not SSE-fresh: `evidenceMapKeys` is never invalidated on `BulletProvenanceRecorded`/`ResumeApproved`/`JobScored`/`ProfileUpdated`/`ProfileImported`, contradicting the E2 scope and Owner Decision #1. |
| Medium | [r3525824187](https://github.com/ebarti/JobHunter/pull/283#discussion_r3525824187) | Zero-usage state hidden: `UsageGroup` returns `null` when empty, so a proof point used nowhere shows no "never used" state in the detail (only a list count badge). |
| Medium | [r3525824493](https://github.com/ebarti/JobHunter/pull/283#discussion_r3525824493) | "When generated" not inspectable: recorded `occurredAt`/`generation`/`scoreVersion`/`lastUsedAt` are never rendered; usage rows show only a label + status badge. |
| Low | [r3525824817](https://github.com/ebarti/JobHunter/pull/283#discussion_r3525824817) | Coverage-usage links discard the recorded `artifactId` and route to `/jobs/$jobId` instead of the covering resume. |
| Low | [r3525825041](https://github.com/ebarti/JobHunter/pull/283#discussion_r3525825041) | Empty-map copy says "No evidence entries match." for genuinely-empty profiles (implies a filter); header meta reads "loading" in the error state. |

None block merge; all three Medium items are audit-surface polish / plan-scope completeness on a read-only surface, and the map is honest about the data it does show.

### Verification (run in isolated worktree, no servers/ports bound)

- `corepack pnpm install --frozen-lockfile` — ok
- `corepack pnpm web:check` — clean
- `corepack pnpm --filter @jobhunter/api-client check` — clean
- `corepack pnpm api:check` — clean (covers qa-seed.ts)
- `corepack pnpm --filter @jobhunter/web test-d` — 11 files / 13 tests pass, no type errors
- `corepack pnpm --filter @jobhunter/web test` — **149 files / 915 tests pass** (parity tests incl. every-event-has-handler green)
- Not run: `pnpm --filter @jobhunter/web e2e` and `pnpm web:storybook:test` (both need running servers; excluded per review constraints). Statically confirmed `apps/web/e2e/tests/evidence-map.spec.ts` and the 5 Storybook states exist.
